api/src/main/java/org/openmrs/module/sync/SyncUtil.java

```diff
@@ -214,19 +213,33 @@ public static NodeList getChildNodes(String incoming) throws Exception {
 		return nodes;
 	}
 	
-	public static void setProperty(Object o, Node n, ArrayList<Field> allFields) throws IllegalArgumentException,
-	                                                                            IllegalAccessException,
-	                                                                            InvocationTargetException {
-		String propName = n.getNodeName();
-		Object propVal = SyncUtil.valForField(propName, n.getTextContent(), allFields, n);
-		
-		log.debug("Trying to set value to " + propVal + " when propName is " + propName + " and context is "
-		        + n.getTextContent());
-		
+	public static void setProperty(Object o, Field field, String xmlFieldName, String xmlFieldValue, String xmlFieldType) throws Exception {
+		Object propVal = null;
+		if (xmlFieldValue != null) {
+			propVal = SyncUtil.valForField(field, xmlFieldValue, xmlFieldType);
+		}
+		log.debug("Trying to set value to " + propVal + " when propName is " + xmlFieldName + " and content is " + xmlFieldValue);
 		if (propVal != null) {

```

Comment:
@cioan / @mogoodrich .  I'm curious about this pre-existing behavior.  Why are we only setting the property if it is non-null?  What about the case where an existing property is non-null, and we are updating it via sync to a null value?  We should add some tests to make sure that behaves correctly and that skipping null values like this is the correct behavior.

Reply:
Not sure I follow, this is a property, not an object.  So, for example, if you have an existing PatientIdentifiertype, with a non-null value for "format".  And then you update the PatientIdentifierType on the local server, setting "format" to null, does this successfully get set to null on the remote server?  I would be shocked if this was not happening, but just seeing that null values were ignored here in the code, it got me worried.  So we should write some tests around this to make sure that properties that are changed from a non-null value to a null value are correctly synced.

Reply:
Yeah, agreed that I'd be surprised if that was not happened, because I figure we would have run into it as a problem before now, but definitely agree worth unit testing and also worth testing what happened on a sync testing dev environment.

Reply:
@cioan - can you add a unit test for what I indicated above?  Basically, test a case where you are changing a property value of an existing object from non-null to null on the child, and then sync the object to the parent.  On the parent, this property should be set to null.

Reply:
Follow-up on this conversation, I have created a new ticket for this issue, which does seem like a bug, but which is pre-existing and not related to this PR / change:  https://issues.openmrs.org/browse/SYNC-357